vamb/__main__.py

```diff
@@ -1248,37 +1299,32 @@ def write_clusters_and_bins(
     clusters: dict[str, set[str]],
     sequence_names: Sequence[str],
     sequence_lens: Sequence[int],
+    to_file: bool = True,
+    unsplit_clusters_file: Optional[TextIO] = None,
+    split_clusters_file: Optional[TextIO] = None,
 ):
     # Write unsplit clusters to file
-    begintime = time.time()
     unsplit_path = Path(base_clusters_name + "_unsplit.tsv")
-    with open(unsplit_path, "w") as file:
-        (n_unsplit_clusters, n_contigs) = vamb.vambtools.write_clusters(
-            file, clusters.items()
-        )
+    split_path = Path(base_clusters_name + "_split.tsv")
+
+    n_unsplit_clusters, _ = write_clusters_table(
+        unsplit_clusters_file, unsplit_path, clusters.items(), to_file
+    )
 
     # Open unsplit clusters and split them
     if binsplitter.splitter is not None:
-        split_path = Path(base_clusters_name + "_split.tsv")
         clusters = dict(binsplitter.binsplit(clusters.items()))
```

Comment:
Consider not instantiating the dict here and using the iterator instead, if possible.